Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/services/GatewayService.java

```diff
@@ -175,10 +185,21 @@ public boolean retrieveDocument(PatientSearchBean patientQuerySearch) {
 
         //Call the NwHIN service to retrieve the document
         DocumentRetrieveResults response = documentRetrieveService.retrieveDocuments(docRetrieve);
-        //set the retrieved document to the UI patient bean 
+
+        //set the retrieved document to the UI patient bean
         if (response.getDocument() != null) {
-            //patientQuerySearch.getSelectedCurrentDocument().setContentType(response.getContentType());
-            patientQuerySearch.getSelectedCurrentDocument().setDocumentContent(response.getDocument());
+            if (response.getContentType().contains("text/xml")) {
+                InputStream xsl = FacesContext.getCurrentInstance().getExternalContext().getResourceAsStream("/WEB-INF/CDA.xsl");
+                InputStream xml = new ByteArrayInputStream(response.getDocument());
+                byte[] convertXmlToHtml = null;
+                if (xsl != null) {
+                    convertXmlToHtml = convertXMLToHTML(xml, xsl);
+                    xsl.close();
+                }
+                patientQuerySearch.getSelectedCurrentDocument().setDocumentContent(convertXmlToHtml);
+            } else {
+                patientQuerySearch.getSelectedCurrentDocument().setDocumentContent(response.getDocument());
```

Comment:
So we're not handling pdf yet.  This seems too broad of a type for the document too.  We'd being applying this cda.xsl to all xmls?  isn't it possible that it could be a c32 or c62 or some other xml document.